httpx/_types.py

```diff
@@ -38,7 +38,7 @@
 ]
 
 HeaderTypes = Union[
-    "Headers", Dict[StrOrBytes, StrOrBytes], Sequence[Tuple[StrOrBytes, StrOrBytes]],
+    "Headers", Dict[AnyStr, AnyStr], Sequence[Tuple[AnyStr, AnyStr]],
```

Comment:
Based off @yeraydiazdiaz's observations it seems this might be closer to what we need:

```suggestion
    "Headers",
    Dict[str, str],
    Dict[bytes, bytes],
    Sequence[Tuple[str, str]],
    Sequence[Tuple[bytes, bytes]],
```

It would work (i.e. tell users "use either str/str or bytes/bytes, but don't mix") even outside of functions :+1:

Reply:
Good call, folks.